eclipse-collections/src/main/java/org/eclipse/collections/impl/bimap/mutable/AbstractMutableBiMap.java

```diff
@@ -778,6 +783,26 @@ public boolean isEmpty()
             return AbstractMutableBiMap.this.isEmpty();
         }
 
+        @Override
+        public K getFirst()
+        {
+            if (AbstractMutableBiMap.this.size() > 0)
+            {
+                return (K) this.toArray()[0];
```

Comment:
@johnjstuart isn't really no other O(1) way to return this information? Frankly transforming the collection to an array is really a costly operation here… same with getLast below.
A solution would be to use `IterableIterate.getFirst` for example… careful with `IterableIterate.getLast` because it would iterate through the whole collection…